graph break on  sparse tensors constructions

laithsakka opened the pull request:
Fix some tests in https://github.com/pytorch/pytorch/issues/119780
sparse_bsc_tensor is not supported 
https://github.com/pytorch/pytorch/pull/117907

Also more about the issue here. 
https://docs.google.com/document/d/1EIb4qG88-SjVFn5TloLERliYdxIu2hwYoAA8skjOVfo/edit

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #120458



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @aakhundov

laithsakka committed:
graph break on sparse_bsc_tensor

[ghstack-poisoned]

ezyang commented:
Adding @yanboliang but I think you are supposed to do something in the dynamo allowed module instead

yanboliang commented on torch/_dynamo/variables/torch.py:
```diff
@@ -409,6 +409,8 @@ def call_function(
                 example_value=None,
             )
+        elif self.value.__name__ == 'sparse_bsc_tensor':
+            unimplemented("sparse_bsc_tensor")
```
The correct way to to this is:
1, Add ```"torch.sparse_bsc_tensor": SkipFunctionVariable,``` to ```manual_torch_name_rule_map``` in ```trace_rules.py```;
2, Remove ```"torch.sparse_bsc_tensor"``` from ```torch_c_binding_in_graph_functions``` in ```trace_rules.py```.
And there are a few similar functions, probably you want have the same treatment:
https://github.com/pytorch/pytorch/blob/2240018c03744ee34ea14ad53481db934c37e384/torch/_dynamo/trace_rules.py#L1989-L1994

laithsakka replied:
thanks for the tip

laithsakka committed:
Update on "graph break on sparse_bsc_tensor"

Fix some tests in https://github.com/pytorch/pytorch/issues/119780
sparse_bsc_tensor is not supported 
https://github.com/pytorch/pytorch/pull/117907

Also more about the issue here. 
https://docs.google.com/document/d/1EIb4qG88-SjVFn5TloLERliYdxIu2hwYoAA8skjOVfo/edit




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

laithsakka committed:
Update on "graph break on  sparse tensors constructions"

Fix some tests in https://github.com/pytorch/pytorch/issues/119780
sparse_bsc_tensor is not supported 
https://github.com/pytorch/pytorch/pull/117907

Also more about the issue here. 
https://docs.google.com/document/d/1EIb4qG88-SjVFn5TloLERliYdxIu2hwYoAA8skjOVfo/edit




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

ezyang reviewed (approved):
okey dokey

laithsakka commented:
I think i expect more tests to be unblocked will take a quick other look on some tests before landing

laithsakka commented:
ahh i see could be of the pending, idk what they have been pending since yesterday though!?

laithsakka committed:
Update on "graph break on  sparse tensors constructions"

Fix some tests in https://github.com/pytorch/pytorch/issues/119780
sparse_bsc_tensor is not supported 
https://github.com/pytorch/pytorch/pull/117907

Also more about the issue here. 
https://docs.google.com/document/d/1EIb4qG88-SjVFn5TloLERliYdxIu2hwYoAA8skjOVfo/edit




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

laithsakka committed:
Update on "graph break on  sparse tensors constructions"

Fix some tests in https://github.com/pytorch/pytorch/issues/119780
sparse_bsc_tensor is not supported 
https://github.com/pytorch/pytorch/pull/117907

Also more about the issue here. 
https://docs.google.com/document/d/1EIb4qG88-SjVFn5TloLERliYdxIu2hwYoAA8skjOVfo/edit




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

laithsakka committed:
Update on "graph break on  sparse tensors constructions"

Fix some tests in https://github.com/pytorch/pytorch/issues/119780
sparse_bsc_tensor is not supported 
https://github.com/pytorch/pytorch/pull/117907

Also more about the issue here. 
https://docs.google.com/document/d/1EIb4qG88-SjVFn5TloLERliYdxIu2hwYoAA8skjOVfo/edit




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

laithsakka committed:
Update

[ghstack-poisoned]

laithsakka committed:
Update

[ghstack-poisoned]

laithsakka committed:
Update

[ghstack-poisoned]

laithsakka commented:
remove on more unexpected success

laithsakka committed:
Update

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge
